### PR TITLE
Added Oven mode yaml scripts and updated ciTests.json file

### PR DESCRIPTION
**Pull request by prakashece**
Oven mode test scripts (YAMLs) are pushed in this PR. Please find the link below to see oven mode test plan
https://github.com/CHIP-Specifications/chip-test-plans/blob/master/src/cluster/mode_oven.adoc

Citests.json is updated

### CODE REVIEW

File: src/app/tests/suites/certification/PICS.yaml [hunk 1]
```diff
@@ -8881,7 +8881,61 @@ PICS:
...
+          LWM.S.C01.Tx
+          #  Oven Mode Cluster
+
+    - label: "Does the device implement the OM cluster as a server?"
+      id: OM.S
+
+    - label: "Does the DUT support testing the failed ChangeToMode command?"
+      id: OM.S.M.CAN_TEST_MODE_FAILURE
+
+    - label:
```
Comment (cecille) [inline src/app/tests/suites/certification/PICS.yaml:8894]: You shouldn't need to add pixits in this file

File: src/app/tests/suites/certification/Test_TC_OM_2_1.yaml [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+    - label: "Step 8: TH reads from the DUT the CurrentMode attribute."
+      PICS: OM.S.A0001
+      command: "readAttribute"
+      attribute: "CurrentMode"
+      response:
+          value: Step6_current_mode_dut
+          constraints:
+              type: int8u
+              minValue: 0
```
[the remark this answers is not in the document]
Comment (cecille) [reply src/app/tests/suites/certification/Test_TC_OM_2_1.yaml:157]: But why is there a check that it's in a truncated int range too?

File: src/app/tests/suites/certification/ci-pics-values [hunk 3]
```diff
@@ -750,6 +750,27 @@ PIXIT.LWM.MODE_CHANGE_OK=1
...
+OM.S.A0002=1
+
+#Feature
+OM.S.F00=1
+
+#Command
+OM.S.C00.Rsp=1
+OM.S.C01.Tx=1
+
+#PIXIT
```
Comment (cecille) [inline src/app/tests/suites/certification/ci-pics-values:768]: These can be removed - pics only in this file. Realistically, this file is terrible anyway, but it'll give us less to clean up.

**Review by cecille — COMMENTED** [review]
Need to sort out the PICS codes before this can go in. Otherwise looks reasonable.

File: src/app/tests/suites/certification/Test_TC_OM_1_2.yaml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+tests:
+    - label:
+          "Step 1: Commission DUT to TH (can be skipped if done in a preceding
+          test)."
+      verification: |
+
+      disabled: true
+
+    - label: "Step 2: TH reads from the DUT the SupportedModes attribute."
```
[the remark this answers is not in the document]
Comment (prakashece) [reply src/app/tests/suites/certification/Test_TC_OM_1_2.yaml:33]: Will work for sure to automate the TCs. The reason is as we are working on tight deadlines for TE2, we couldn't put much time on it.

File: src/app/tests/suites/certification/Test_TC_OTCCM_2_1.yaml [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+    - label: "Step 3: TH reads from the DUT the CurrentMode attribute."
+      PICS: OTCCM.S.A0001
+      command: "readAttribute"
+      attribute: "CurrentMode"
+      response:
+          saveAs: old_current_mode_dut
+          constraints:
+              type: int8u
+              minValue: 0
+              maxValue: 254
```
Comment (cecille) [inline src/app/tests/suites/certification/Test_TC_OTCCM_2_1.yaml:68]: why is this 254 and not 255? ditto below

File: src/app/tests/suites/certification/Test_TC_OTCCM_2_1.yaml [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+          values:
+              - name: "nodeId"
+                value: nodeId
+
+    - label: "Step 2: TH reads from the DUT the SupportedModes attribute."
+      PICS: OTCCM.S.A0000
+      command: "readAttribute"
+      attribute: "SupportedModes"
+      response:
+          saveAs: "suppoted_mode_dut"
```
Comment (cecille) [inline src/app/tests/suites/certification/Test_TC_OTCCM_2_1.yaml:53]: there's a spelling error in the variable name here.  Is this used anywhere?

**Review by cecille — APPROVED** [review]

File: src/app/tests/suites/certification/Test_TC_OTCCM_2_1.yaml (hunk 6 above)
Comment (prakashece) [reply src/app/tests/suites/certification/Test_TC_OTCCM_2_1.yaml:53]: nope, not used, but still I corrected it

**Review by jmartinez-silabs — APPROVED** [review]

**Review by jmartinez-silabs — COMMENTED** [review]
Seems like the cluster is not enabled on the all-app-cluster for the tests.

**Comment by andy31415** [comment]
closed since #31547 claims to be its replacement

**Pull request closed without merging**